compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -166,21 +166,24 @@ class AnonymousFunctionMissingParamType(param: untpd.ValDef,
                                         (using Context)
 extends TypeMsg(AnonymousFunctionMissingParamTypeID) {
   def msg(using Context) = {
-    val ofFun =
+    val paramDescription =
       if param.name.is(WildcardParamName)
-          || (MethodType.syntheticParamNames(tree.args.length + 1) contains param.name)
-      then i"\n\nIn expanded function:\n$tree"
+          || param.name.is(ContextFunctionParamName)
+          || MethodType.syntheticParamNames(tree.args.length + 1).contains(param.name)
+      then i"\nin expanded function:\n  $tree"
       else ""
 
     val inferred =
-      if (inferredType == WildcardType) ""
-      else i"\n\nPartially inferred type for the parameter: $inferredType"
+      if inferredType == WildcardType then ""
+      else i"\nWhat I could infer was: $inferredType"
```

Comment:
```suggestion
      else i"\nWhat I could infer for the parameter was: $inferredType"
```

It is not clear if this type is that of the parameter, or the whole function, which can be made even more confusing in case it infers a function type for the parameter !

This is why we had opted for verbosity, but I agree our solution was a bit unelegant

Reply:
I think it is clear from the context: "coould not infer type for for parameter ${param.name} ...", "what i could infer was: ...". Also, we do give the full expected type below as well.